poi/src/main/java/org/apache/poi/hssf/model/DrawingManager2.java

```diff
@@ -19,21 +19,156 @@ Licensed to the Apache Software Foundation (ASF) under one or more
+
+        EscherDggRecord dggRecord = dggContainer.getChildById(EscherDggRecord.RECORD_ID);
+        if (dggRecord == null) {
+            dggRecord = createEscherDggRecord();
+            dggContainer.addChildRecord(dggRecord);
+        }
+
+        EscherContainerRecord bStoreContainer = dggContainer.getChildById(EscherContainerRecord.BSTORE_CONTAINER);
+        if (bStoreContainer == null) {
+            // Create an empty store and use that. An empty store doesn't cause any problems, but makes the programmatic
+            //  model easier to understand.
+            bStoreContainer = new EscherContainerRecord();
+            bStoreContainer.setRecordId(EscherContainerRecord.BSTORE_CONTAINER);
+            bStoreContainer.setOptions((short) (0xF));
+
+            List<EscherRecord> childRecords = dggContainer.getChildRecords();
+            childRecords.add(1, bStoreContainer);
+            dggContainer.setChildRecords(childRecords);
+        }
+
+        return new DrawingManager2(dggRecord, bStoreContainer);
+    }
+
+    /**
+     * Creates a primary drawing group record.
+     */
+    private static DrawingGroupRecord createDrawingGroupRecord() {
+        EscherContainerRecord dggContainer = new EscherContainerRecord();
+        EscherDggRecord dgg = createEscherDggRecord();
+        EscherOptRecord opt = new EscherOptRecord();
+        EscherSplitMenuColorsRecord splitMenuColors = new EscherSplitMenuColorsRecord();
+
+        dggContainer.setRecordId(EscherContainerRecord.DGG_CONTAINER);
+        dggContainer.setOptions(EscherContainerRecord.DGG_CONTAINER);
+
+        EscherContainerRecord bStoreContainer = new EscherContainerRecord();
+        bStoreContainer.setRecordId(EscherContainerRecord.BSTORE_CONTAINER);
+        bStoreContainer.setOptions((short) (0xF));
+
+        opt.setRecordId(EscherOptRecord.RECORD_ID);
+        opt.addEscherProperty(new EscherBoolProperty(EscherPropertyTypes.TEXT__SIZE_TEXT_TO_FIT_SHAPE, 524296));
+        opt.addEscherProperty(new EscherRGBProperty(EscherPropertyTypes.FILL__FILLCOLOR, 0x08000041));
+        opt.addEscherProperty(new EscherRGBProperty(EscherPropertyTypes.LINESTYLE__COLOR, 134217792));
+        splitMenuColors.setRecordId(EscherSplitMenuColorsRecord.RECORD_ID);
+        splitMenuColors.setOptions((short) 0x0040);
+        splitMenuColors.setColor1(0x0800000D);
+        splitMenuColors.setColor2(0x0800000C);
+        splitMenuColors.setColor3(0x08000017);
+        splitMenuColors.setColor4(0x100000F7);
+
+        dggContainer.addChildRecord(dgg);
+        dggContainer.addChildRecord(bStoreContainer);
+        dggContainer.addChildRecord(opt);
+        dggContainer.addChildRecord(splitMenuColors);
+
+        DrawingGroupRecord drawingGroup = new DrawingGroupRecord();
+        drawingGroup.addEscherRecord(dggContainer);
+        return drawingGroup;
+    }
+
+    private static EscherDggRecord createEscherDggRecord() {
+        EscherDggRecord dgg = new EscherDggRecord();
```

Comment:
not the biggest fan of the inconsistent naming from dgg to dggRecord, would prefer dggRecord in all cases